### PR TITLE
Added wrapped rwops data structure, fixed surface-format-format return value bug, added a method to see surface pixel format type for debug purposes

### PR DESCRIPTION
**Pull request by Failproofshark**
Added a way to create wrapped RWOPs structures which is helpful for some added functionality in cl-sdl2-image. Also surface-format-format returned a struct instead of an integer which gives the enum value of a surface

### CODE REVIEW

File: src/surface.lisp [hunk 1]
```diff
@@ -17,8 +17,11 @@
   (c-ref surface sdl2-ffi:sdl-surface :pitch))
 
 (defun surface-format-format (surface)
-  (enum-key '(:enum (sdl-pixel-format))
-            (c-ref surface sdl2-ffi:sdl-surface :format :format)))
+  (c-ref (c-ref surface
+                sdl2-ffi:sdl-surface
+                :format *)
+         sdl2-ffi:sdl-pixel-format
+         :format))
```
Comment (rpav) [inline src/surface.lisp]: Er, why was this changed?  You should not need nested `C-REF`, and this should return the *keyword* not just the numeric value.

**Comment by Failproofshark** [comment]
I'm not really sure why but when I tested it initially the function returned some sort of structure rather than the keyword. I can roll back and just make a commit for the rwops struct

**Comment by rpav** [comment]
Yeah, we had `SURFACE-FORMAT` return the "format" structure itself, then `SURFACE-FORMAT-FORMAT` returning the keyword (a la `(c-ref .. surface :format :format)`).  Both should still exist; I think calling them `SURFACE-FORMAT` and `SURFACE-FORMAT-STRUCT` would probably be clearer, but ah well.

Having the keyword returned here is probably most-useful, since it can be used interchangeably, and the primary reason for calling the function is probably checking visually what the format <i>is</i>.
